File: jdroid-java-webapp-core/src/main/java/com/jdroid/javaweb/sentry/SentryAppModule.java
package com.jdroid.javaweb.sentry;

import com.getsentry.raven.Raven;
import com.getsentry.raven.RavenFactory;
import com.jdroid.javaweb.application.AbstractAppModule;
import com.jdroid.javaweb.application.Application;
import com.jdroid.javaweb.config.CoreConfigParameter;

public class SentryAppModule extends AbstractAppModule {

	public static final String MODULE_NAME = SentryAppModule.class.getName();

	public static SentryAppModule get() {
		return (SentryAppModule)Application.get().getAppModule(MODULE_NAME);
	}

	private Raven raven;

	public Raven getRaven() {
		if (raven == null) {
			raven = RavenFactory.ravenInstance(Application.get().getRemoteConfigLoader().getString(CoreConfigParameter.SENTRY_DSN));
		}
		return raven;
	}

	@Override
	public void onContextDestroyed() {
		if (raven != null) {
			getRaven().closeConnection();
		}
	}
}

File: jdroid-java-webapp-core/src/main/java/com/jdroid/javaweb/sentry/SentryAppModule.kt
package com.jdroid.javaweb.sentry

import com.getsentry.raven.Raven
import com.getsentry.raven.RavenFactory
import com.jdroid.javaweb.application.AbstractAppModule
import com.jdroid.javaweb.application.Application
import com.jdroid.javaweb.config.CoreConfigParameter

class SentryAppModule : AbstractAppModule() {

    companion object {

        val MODULE_NAME: String = SentryAppModule::class.java.name

        fun get(): SentryAppModule {
            return Application.get().getAppModule(MODULE_NAME) as SentryAppModule
        }
    }

    val raven: Raven by lazy {
        RavenFactory.ravenInstance(Application.get().remoteConfigLoader.getString(CoreConfigParameter.SENTRY_DSN))
    }

    override fun onContextDestroyed() {
        raven.closeConnection()
    }
}
